salt/modules/netbox.py

```diff
@@ -82,28 +90,90 @@ def _strip_url_field(input_dict):
     return input_dict
 
 
-def filter(app, endpoint, **kwargs):
+def _dict(iterable):
+    if iterable:
+        return dict(iterable)
+    else:
+        return {}
+
+
+def _add(app, endpoint, payload):
+    '''
+    POST a payload
+    '''
+    nb = _nb_obj(auth_required=True)
+    try:
+        return getattr(getattr(nb, app), endpoint).create(**payload)
+    except RequestError as e:
+        log.error("{}, {}, {}".format(e.req.request.headers, e.request_body, e.error))
+        return False
+
+
+def slugify(value):
+    ''''
+    Slugify given value.
+    Credit to Djangoproject https://docs.djangoproject.com/en/2.0/_modules/django/utils/text/#slugify
+    '''
+    value = re.sub(r'[^\w\s-]', '', value).strip().lower()
+    return re.sub(r'[-\s]+', '-', value)
+
+
+def _get(app, endpoint, id=None, auth_required=False, **kwargs):
+    '''
+    Helper function to do a GET request to Netbox.
+    Returns the actual pynetbox object, which allows manipulation from other functions.
+    '''
+    nb = _nb_obj(auth_required=auth_required)
+    if id:
+        item = getattr(getattr(nb, app), endpoint).get(id)
+    else:
+        item = getattr(getattr(nb, app), endpoint).get(**salt.utils.args.clean_kwargs(**kwargs))
+    return item
+
+
+def _if_name_unit(if_name):
+    if_name_split = if_name.split('.')
+    if len(if_name_split) == 2:
+        return if_name_split
+    return if_name, '0'
+
+
+def filter_(app, endpoint, **kwargs):
```

Comment:
We like to avoid the use of `**kwargs` because it's unclear to the end user what it supposed to be passed here. We either need this made into an explicit variable or an explanation of what's expected here inserted into the function documentation, please.

Reply:
Done.